Repeated warning on master compiling

Using waf on macos to compile I get this warning repeating...

~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~
1 warning generated.

[ 578/1207] Compiling ArduCopter/switches.cpp
In file included from ../../ArduCopter/precision_landing.cpp:7:
In file included from ../../ArduCopter/Copter.h:115:
../../libraries/AP_Frsky_Telem/AP_Frsky_Telem.h:130:65: warning: expression result unused [-Wunused-value]
void set_is_flying(bool is_flying) { is_flying ? (_ap.value | AP_ISFLYING_FLAG) : (_ap.value & (~AP_ISFLYING_FLAG)); }
~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~
1 warning generated.

[ 579/1207] Compiling ArduCopter/system.cpp
In file included from ../../ArduCopter/radio.cpp:3:
In file included from ../../ArduCopter/Copter.h:115:
../../libraries/AP_Frsky_Telem/AP_Frsky_Telem.h:130:65: warning: expression result unused [-Wunused-value]
void set_is_flying(bool is_flying) { is_flying ? (_ap.value | AP_ISFLYING_FLAG) : (_ap.value & (~AP_ISFLYING_FLAG)); }
~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~
1 warning generated.

[ 580/1207] Compiling ArduCopter/takeoff.cpp
In file included from ../../ArduCopter/sensors.cpp:3:
In file included from ../../ArduCopter/Copter.h:115:
../../libraries/AP_Frsky_Telem/AP_Frsky_Telem.h:130:65: warning: expression result unused [-Wunused-value]
void set_is_flying(bool is_flying) { is_flying ? (_ap.value | AP_ISFLYING_FLAG) : (_ap.value & (~AP_ISFLYING_FLAG)); }

perhaps related to this ?
https://github.com/ArduPilot/ardupilot/pull/4892#discussion_r80533534

--- a/libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp
+++ b/libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp
@@ -907,3 +907,12 @@ void AP_Frsky_Telem::calc_gps_position(void)
         _gps.speed_in_centimeter = 0;
     }
 }
+
+void AP_Frsky_Telem::set_is_flying(bool is_flying)
+{
+    if (is_flying) {
+        _ap.value |= AP_ISFLYING_FLAG;
+    } else {
+        _ap.value &= ~AP_ISFLYING_FLAG;
+    }
+}

--- a/libraries/AP_Frsky_Telem/AP_Frsky_Telem.h
+++ b/libraries/AP_Frsky_Telem/AP_Frsky_Telem.h
@@ -127,7 +127,7 @@ public:
     void update_control_mode(uint8_t mode) { _ap.control_mode = mode; }
 
     // update whether we're flying (used for Plane)
-    void set_is_flying(bool is_flying) { is_flying ? (_ap.value | AP_ISFLYING_FLAG) : (_ap.value & (~AP_ISFLYING_FLAG)); }
+    void set_is_flying(bool is_flying);
  
     // update error mask of sensors and subsystems. The mask uses the
     // MAV_SYS_STATUS_* values from mavlink. If a bit is set then it

